### PR TITLE
feat: add isOldAsyncAPIDocument helper

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

- add `x-parser-api-version` to have info about parser-api used in AsyncAPI model
- `isOldAsyncAPIDocument` helper function to check if document use old api
- improve `isAsyncAPIDocument`
- add unit tests

cc @jonaslagoni as you want :)

### CODE REVIEW

**Review by jonaslagoni — COMMENTED** [review]
Do you want to add a converter as well? I.e. from old parser document to new version? The code you send works:

```ts

        // Is from old parser
        const parsedJSON = input.json();
        const detailed = createDetailedAsyncAPI(parsedJSON, parsedJSON);
        doc = createAsyncAPIDocument(detailed);
```

**Comment by derberg** [comment]
one cent from me, please update readme where we document all custom parser extensions

**Comment by magicmatatjahu** [comment]
@jonaslagoni I can add but it won't work in 100% cases.

File: src/document.ts [hunk 1]
```diff
@@ -32,7 +34,25 @@ export function toAsyncAPIDocument(maybeDoc: unknown): AsyncAPIDocumentInterface
...
+  if (!maybeDoc) {
+    return false;
+  }
+  if (maybeDoc instanceof AsyncAPIDocumentV2 || maybeDoc instanceof AsyncAPIDocumentV3) {
+    return true;
+  }
+  if (maybeDoc && typeof (maybeDoc as AsyncAPIDocumentInterface).json === 'function') {
+    const versionOfParserAPI = (maybeDoc as AsyncAPIDocumentInterface).json()[xParserApiVersion];
+    return versionOfParserAPI === 1;
+  }
```
Comment (jonaslagoni) [inline src/document.ts:46]: You dont want to reuse `isOldAsyncAPIDocument` here? 🤔

**Review by jonaslagoni — COMMENTED** [review]
Just one comment, but other then that it LGTM 👍

File: src/document.ts (hunk 1 above)
Comment (magicmatatjahu) [reply src/document.ts:46]: why?

Comment (magicmatatjahu) [reply src/document.ts:46]: We cannot reuse because we need to know if `maybeDoc` has `json()` method.

**Review by derberg — APPROVED** [review]

**Pull request merged**